providers/implementations/kdfs/tls1_prf.c

```diff
@@ -250,16 +252,28 @@ static int kdf_tls1_prf_set_ctx_params(void *vctx, const OSSL_PARAM params[])
     if ((p = OSSL_PARAM_locate_const(params, OSSL_KDF_PARAM_SEED)) != NULL) {
         for (; p != NULL; p = OSSL_PARAM_locate_const(p + 1,
                                                       OSSL_KDF_PARAM_SEED)) {
-            const void *q = ctx->seed + ctx->seedlen;
-            size_t sz = 0;
-
-            if (p->data_size != 0
-                && p->data != NULL
-                && !OSSL_PARAM_get_octet_string(p, (void **)&q,
-                                                TLS1_PRF_MAXBUF - ctx->seedlen,
-                                                &sz))
-                return 0;
-            ctx->seedlen += sz;
+            if (p->data_size != 0 && p->data != NULL) {
+                const void *val;
+                size_t sz;
+                unsigned char *seed;
+                size_t seedlen;
+                int err;
+
+                if (!OSSL_PARAM_get_octet_string_ptr(p, &val, &sz))
+                    return 0;
+
+                seedlen = safe_add_size_t(ctx->seedlen, sz, &err);
+                if (err)
+                    return 0;
+
+                seed = OPENSSL_realloc(ctx->seed, seedlen);
```

Comment:
Shouldn't there be a size check here?  65535 being a hard limit.

Reply:
I'm not sure if there is a limit for the entire seed value at this level; the limit for context is defined in the [caller](https://www.rfc-editor.org/rfc/rfc5705#section-4) where the concatenated seed value may exceed 65535, where client_random and server_random are included.

Reply:
We probably should check the context length if there's a limit.  Overflowing 2¹⁶ is seems more likely than the need for size_t safe math operations :grin:

Reply:
It is checked [here](https://github.com/openssl/openssl/pull/22465/files#diff-4dbb9df153774f168e39eeece254b388123e5ded70bd5a34fa1dfa11a74ad10fR470) in the caller.

Reply:
Thanks.